frontend/src/components/BetaTesterForm.vue

```diff
@@ -59,14 +59,19 @@ export default {
     validators() {
       return validators
     },
+    measures() {
+      const userCanteen = this.$store.state.loggedUser ? this.$store.state.userCanteens[0] : null
+      const diagnostics = userCanteen ? userCanteen.diagnostics : this.$store.getters.getLocalDiagnostics()
+      return diagnostics.find((x) => x.year === 2020) || {}
+    },
   },
   methods: {
     subscribeBetaTester() {
       this.$refs.form.validate()
       if (!this.formIsValid) return
 
       this.$store
-        .dispatch("subscribeBetaTester", this.formData)
+        .dispatch("subscribeBetaTester", { ...this.formData, ...{ measures: this.measures } })
```

Comment:
Peut être une écriture un peu plus claire comme :
```
this.formData["measures"] = this.measures;
this.$store
.dispatch("subscribeBetaTester", this.formData)
```

Reply:
Pareil ça s'est pas joué à grande chose dans ma tête ^^ 

Il y a une petite différence entre les deux options : en utilisant le spread (`...`) on laisse `formData` inchangé. Vu que notre template est bindé à `formData` et que finalement l'objet passé au dispatch n'a pas vocation à être utilisé ailleurs, je me suis dit que ce serait plus propre de laisser `formData` tel qu'il est et juste créer un objet temporaire pour le dispatch.

Reply:
Dans ce cas est-ce qu'on peut pas plutôt faire :
`.dispatch("subscribeBetaTester", this.formData, this.measures)`
et dans le store quelque chose comme :
```
subscribeBetaTester(context, canteenInfo, latestDiagnostic) {
    let payload = canteenInfo;
    payload["latestDiagnostic"] = latestDiagnostic;
    return fetch(...);
```
Après tout c'est la responsabilité du store de formatter le payload comme il faut.

Reply:
En effet c'est une option, après je ne suis pas sur que ça soit la responsabilité du store d'assembler le payload mais plutôt d'envoyer la requête et de muter le state :thinking:  Disons que si c'était le cas, le state obtiendrait le dernier diagnostic lui même - ce qui est possible mais qui ajoute de la responsabilité à une zone qui s'en charge déjà de pas mal de choses. Le composant chargé de cette opération me semble un endroit plus logique pour créer le payload.

Reply:
L'argument en faveur que ce soit le store qui gère ça c'est que si on change le contrat côté api, par exemple si on veut un payload `{ basicInfos: ..., diagnostic: ... }` y'a pas besoin de chercher dans le code tous les appels à subscribeBetaTester, c'est juste au niveau du store que ça se passe.

Reply:
Ça marche pour un changement de nom d'une propriété au tout premier niveau (en fait juste la propriété mesure), mais c'est plus le cas si le service change par exemple pour recevoir un tableau de diagnostics au lieu d'un seul, ou si le nom d'une propriété un niveau au dessous change. Après c'est pour ça aussi qu'on a le _v1_ sur `/api/v1` pour les breaking changes de l'API.

Juste pour être sur de bien comprendre le souci initial, ce qui te gène à la base c'est de ne pas avoir une écriture plus claire / moins exotique que le spread `...` ?

Reply:
Surtout ça oui.